### DISCUSSION TITLE
Open new screen an Form1 (update mode) on ID = say 5

### DISCUSSION THREAD

**Discussion by Townieeee** [Budibase/budibase#13796] [Help] [answered]
Please can someone tell me the steps from a button to Open screen1 which has a Form1 (in update mode) on ID = say 5.
I have a  state variable also -  vIDState set as 5 say. (which is the rec I want to go to and be in update mode)

**Comment by mikesealey** [comment, marked as answer]
Hi @Townieeee 

Whilst App State is an incredibly powerful part of Budibase, we can achieve what you're aiming for more more simply. I'll walk you through a quick demonstration of how I'd take your puzzle, and answer any questions you've got. As a little extra I'll attach my app-export for you to take a look at the internals of as well.

- Firstly, I created a new app, and used the Budibase sample data. 
- Next, I created a screen with the following component-tree:
![image](https://github.com/Budibase/budibase/assets/39269380/c3917fee-d565-411a-a3e2-8021b9626855)

This component tree In brief: 

- My `Data Provider ` gets the rows from the _Employees_ table
- The repeater then iterates over that table, and the repeater-block gives me access to one row of the results from the data provider. You only need one repeater-block in your component tree, Budibase will make a new one for each row.
- I've then added in a container which holds a stat-card. My stat card is set to show values about _this_ row using the bindings

![image](https://github.com/Budibase/budibase/assets/39269380/61973bd8-b87e-4df6-8f87-63883d2e746c)

The result of this is that every employee in my table gets a Stat Card created for them
![image](https://github.com/Budibase/budibase/assets/39269380/50227bb4-49c3-4f2f-b367-2681065fb639)

- Here's where we'll really get to the problem you're trying to tackle. I've created a new page called `/employees/:id` (Note the colon `:` ) this signals that what comes after it is a _parameter_ - in my case a row ID.
- On the screen `/employees/:id` I used a `Form Block` set with `Data: Employees` and `Type: Update`
- Now my form knows which table to update, but which row? Click to open the `Row ID` bindings, and go to URL and chose `id`
![image](https://github.com/Budibase/budibase/assets/39269380/f94e1502-9371-4f22-87e4-b1ca5c174547)
![image](https://github.com/Budibase/budibase/assets/39269380/dce55c3c-5e0b-43d4-a5cf-cafa4ba611ae)
![image](https://github.com/Budibase/budibase/assets/39269380/865261c4-aefc-4fd7-be4e-d64cd5b6b54d)

- Remember that container I put the `Stat Card` in? I've set an `On Click` action to `Navigate To`  with the path set to `/employees/{{ New Repeater Block.Employees._id }}` which passes this row id to the Screen (and form) where you'll edit a row. 
![image](https://github.com/Budibase/budibase/assets/39269380/24baa630-5c33-403b-a658-7b37270a8a77)

Now, when I go to test this I can click on anyone in my list of Stat Cards and go to a form that lets me update that employee's information.

One slight caveat to remember: When I created the `/employees/:id` screen, it automatically added itself to the navigation bar, which means I can visit the screen without setting the `:id`, which by default shows this:
![image](https://github.com/Budibase/budibase/assets/39269380/5f11b2e3-b3a7-4d79-a1a6-627dbf47c87d)

But really, we don't want that at all! In the builder, click on the navigation bar, and delete the link that takes us to `/screen/:id`
![image](https://github.com/Budibase/budibase/assets/39269380/d2413298-652b-41dc-95f9-1423f97937ec)

I hope this helps get you on your way. As promised, here's the app-export for you to take a look at: [13796-export-1717073240973.tar.gz](https://github.com/Budibase/budibase/files/15500118/13796-export-1717073240973.tar.gz) 

If there's anything else you need help with, you may also find answers in the [Discord Server](https://discord.com/invite/ZepTmGbtfF) and [The Documentation](https://docs.budibase.com/docs/what-is-budibase).

**Reply by Townieeee** [reply]
Thank you so much. I look forward to trying this exactly like you did.
I am very new to this so please tell me how I can see your export as an App?
I have Windows10 so unzipped by downloading 7zip then extracted 3 files from your file.
I now have to guess how to import your "App" plus "data" I guess - into Budibase. I was unsuccessful at finding somewhere that would import/restore this. At this point I feel like I am ready to try drive this car but cannot find the keyhole to even get into the car to try.

**Reply by mikesealey** [reply]
[Here's the page in the documentation](https://docs.budibase.com/docs/export-and-import-apps) specifically looking at importing and exporting apps.

**Reply by Townieeee** [reply]
thx again Mike - I got the import in and it really did help me understand better.
So in your example you are coming from a place of many rows to choose from and if you chose the 5th row (lets say was id=5) then you land on the update form (id=5) which is great - but what if you came from a single create form (public, no sign in, Just collect say EmplNo, and HoursWorked) with a submit button which saves to a CollectDataTable and you want to "Get the last saved ID" of this table and open that record on an UPDATE form? I am guessing that I could still use this new learned method of passing the id thru the URL (after figuring out how to "Get the last saved ID")
